storage/src/mmr/operation_array.rs

```diff
@@ -0,0 +1,161 @@
+//! An `Array` implementation for operations applied to a `MutableMmr` K/V store, making them
+//! storable in a `fixed::Journal`.
+
+use commonware_codec::{Codec, Error as CodecError, Reader, SizedCodec, Writer};
+use commonware_utils::{hex, Array};
+use std::{
+    cmp::{Ord, PartialOrd},
+    fmt::{Debug, Display},
+    hash::Hash,
+    ops::Deref,
+};
+use thiserror::Error;
+
+/// Errors returned by `Operation` functions.
+#[derive(Error, Debug)]
+pub enum Error<K: Array, V: Array> {
+    #[error("invalid length")]
+    InvalidLength,
+    #[error("invalid key: {0}")]
+    InvalidKey(<K as Array>::Error),
+    #[error("invalid value: {0}")]
+    InvalidValue(<V as Array>::Error),
+}
+
+/// The types of operations that change a key's state in the store.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub enum Type<V: Array> {
+    /// Indicates the key no longer has a value.
+    Deleted,
+
+    /// Indicates the key now has the wrapped value.
+    Update(V),
+}
+
+/// An `Array` implementation for operations applied to a `MutableMmr` K/V store.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[repr(transparent)]
+pub struct Operation<K: Array, V: Array> {
+    pub data: Vec<u8>,
+    _phantom: std::marker::PhantomData<(K, V)>,
+}
+
+impl<K: Array, V: Array> Operation<K, V> {
+    pub fn new(key: K, op_type: Type<V>) -> Self {
+        let mut data = Vec::with_capacity(Self::LEN_ENCODED);
+        data.extend_from_slice(&key);
+        match op_type {
+            Type::Deleted => {
+                data.push(0);
+                data.extend(vec![0u8; V::LEN_ENCODED]);
```

Comment:
I think we could make this a const to avoid allocating memory for it each time we write a delete.

Reply:
Unfortunately can't make a const since it's a vector (& if I try to use an array it complains that we're referencing a generic derived value for ENCODED_LEN)

Reply:
Did some digging, and we can do `data.resize(Self::LEN_ENCODED, 0)` I think.